translate/src/tools/identify_project_kind.rs

```diff
@@ -0,0 +1,72 @@
+use std::{fmt::Display, path::Path};
+
+use harvest_ir::Representation;
+
+use super::{MightWriteContext, MightWriteOutcome, RunContext, Tool, load_raw_source::RawSource};
+
+pub enum ProjectKind {
+    Library,
+    Executable,
+}
+
+impl Display for ProjectKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ProjectKind::Library => write!(f, "Library"),
+            ProjectKind::Executable => write!(f, "Executable"),
+        }
+    }
+}
+
+impl Representation for ProjectKind {
+    fn name(&self) -> &'static str {
+        "KindAndName"
+    }
+
+    fn materialize(&self, path: &Path) -> std::io::Result<()> {
+        std::fs::create_dir(path)?;
+        match self {
+            ProjectKind::Library => std::fs::write(path.join("library"), []),
+            ProjectKind::Executable => std::fs::write(path.join("executable"), []),
+        }
+    }
```

Comment:
Why use the file name to communicate the type rather than file contents? It seems to me you could skip implementing `materialize` entirely and rely on the default impl (which dumps the `Display` output into a file).

Reply:
Oh... yes, will do. Originally I was getting both the project kind and the name, and storing the name in the file. That turns out to be more tricky to get reliably and also not immediately used anyway, so I got rid of it for now. Eventually, there are also projects that output both a library and an executable and might have different names for each.

But for now, just the default implementation is best indeed.